pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompactmerge/UpsertCompactMergeTaskGenerator.java

```
@@ -462,4 +464,10 @@ protected String getSegmentCrcList(List<SegmentMergerMetadata> segmentMergerMeta
         segmentMergerMetadataList.stream().map(x -> String.valueOf(x.getSegmentZKMetadata().getCrc()))
             .collect(Collectors.toList()), ",");
   }
+
+  @VisibleForTesting
+  protected long getMaxCreationTimeMillis(List<SegmentMergerMetadata> segmentMergerMetadataList) {
```

Should we fail instead of returning -1? Also, we can be explicit by naming this getMaxZKCreationTimeMillis.

---

I wanted to keep the same behaviour as getCreationTime() which also returns -1 if it's not present. 
This is why I am returning -1 and then added a check in executor to check for time > 0.